rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,186 @@
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_spaces_served_by_swh_use import (
+    get_spaces_served_by_swh_use,
+)
+from rct229.schema.config import ureg
+from rct229.schema.schema_enums import SchemaEnums
+from pint import Quantity
+from rct229.utils.assertions import assert_
+from rct229.utils.jsonpath_utils import find_exactly_one_with_field_value, find_all
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.utility_functions import find_exactly_one_schedule
+
+SERVICE_WATER_HEATING_USE_UNIT = SchemaEnums.schema_enums[
+    "ServiceWaterHeatingUseUnitOptions"
+]
+
+REQUIRED_USE_UNIT = [
+    SERVICE_WATER_HEATING_USE_UNIT.POWER,
+    SERVICE_WATER_HEATING_USE_UNIT.VOLUME,
+]
+
+
+def get_energy_required_to_heat_swh_use(
+    swh_use: dict, rmd: dict, building_segment: dict
+) -> dict[str, Quantity]:
+    """
+    This function calculates the total energy required to heat the SWH use over the course of a year in btu.  Note - this function does not work for service water heating uses with use_units == "OTHER".  In this case, it will return 0.
+
+    Parameters
+    ----------
+    swh_use: dict, service_water_heating_uses
+    rmd: dict, RMD at RuleSetModelDescription level
+    building_segment: dict, building_segment
+
+    Returns
+    ----------
+    energy_required_by_space: A dict where the keys are space_ids and values are the total energy required to heat the swh_use for that space.  If a swh_use is not assigned to any spaces, the key will be "NO_SPACES_ASSIGNED"
+
+    """
+
+    hourly_schedule_id = swh_use.get("use_multiplier_schedule")
+    hourly_schedule = (
+        find_exactly_one_schedule(rmd, hourly_schedule_id)
+        if hourly_schedule_id is not None
+        else None
+    )
+
+    distribution_system_id = swh_use.get("served_by_distribution_system", "")
```

Comment:
This is getting into details - I think the key question we should ask ourself is: do we need distribution system in this function?
Based on the RDS, the `distribution_system` is used to get three values:
1. `supply_temperature` when `swh_use.use_units` is volume based units
2. `inlet_T_sched` when `swh_use.use_units` is volume based units
3. `drain_heat_recovery_efficiency` when `swh_use.is_heat_recovered_by_drain`

So in those cases, I would say this is absolutely needed, but not other cases.

Based on that, I will want to know if we indeed need the distribution system by:
```python
use_units = getattr_(swh_use, "swh_use", "use_units")
drain_heat_recovery_efficiency = swh_use.get(...

if use_units in ["VOLUME", ",...] or drain_heat_recovery_efficiency:
    distribution_system_id = getattr_(swh_use, "swh_use", "served_by_distribution_system")
```